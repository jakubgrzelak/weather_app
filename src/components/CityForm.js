import React from 'react';
import { withFormik } from 'formik';
import LoadAnimation from './LoadingAnimation/index';
import cityList from '../assets/city.list.min.json'
import '../styles/components/CityForm.scss'

const MyForm = props => {
  const {
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting
  } = props;

  const formStyle = props.isCityChoosen ? "city-form-inline" : "city-form" 

  return (
    <div>
      <div>
      </div>
      <form onSubmit={handleSubmit} className={formStyle}>
      { isSubmitting && <LoadAnimation />}
        <div className="error-message">
          {errors.name && touched.name && errors.name}
        </div>
        <input
          type="text"
          placeholder='Type city'
          onChange={handleChange}
          onBlur={handleBlur}
          value={values.name}
          name="name"
          className="form-control custom-input" 
        />
        <button className="btn btn-md btn-secondary custom-submit" type="submit">Submit</button>
      </form>
    </div>

  );
};

const MyEnhancedForm = {
  mapPropsToValues: () => ({ name: '' }),

  // Custom sync validation
  validate: values => {
    const errors = {};

    if (!values.name) {
      errors.name = 'Required';
    }

    return errors;
  },

  handleSubmit: (values, { setSubmitting, setFieldError ,props }) => {
    setSubmitting(true)
    const cityName = values.name.trim();
    const matchedCities = cityList.filter(city => city.name === cityName)
    if (matchedCities.length === 0) {
      setFieldError('name', 'City not found')
    } else if (matchedCities.length === 1) {
      props.saveOptionalCities(matchedCities);
      props.onCityClick(matchedCities[0].id)
    } else {
      props.saveOptionalCities(matchedCities);
      props.showCitiesModal();
    }
    setSubmitting(false);
  },

  displayName: 'BasicForm',
}

export default withFormik(MyEnhancedForm)(MyForm)