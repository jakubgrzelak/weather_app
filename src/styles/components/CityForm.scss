.city-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;  

  .custom-input {
    border-radius: 10px;
    width: 300px;
    border-radius: 5px;
  }

  .custom-submit {
    width: 200px;
    margin-top: 10px;
  }

  .error-message {
    height: 20px;
    margin-bottom: 10px;
    color: red
  }
}

.city-form-inline {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;

  .custom-input {
    border-radius: 10px;
    width: 300px;
    border-radius: 5px;
    margin-right: 8px
  }
}