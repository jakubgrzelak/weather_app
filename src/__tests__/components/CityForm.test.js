import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import CityForm from '../../components/CityForm';

test('should render city form correctly', () => {
  const wrapper = shallow(<CityForm />);
  expect(wrapper.debug()).toMatchSnapshot();
});

test('should render ExpenseForm correctly with expense data', () => {
  const wrapper = mount(<CityForm values={{ name: 'London'}} />);
  expect(wrapper.debug()).toMatchSnapshot();
});

// test('should set description on input change', () => {
//   const value = 'London';
//   const wrapper = mount(<CityForm />);
//   wrapper.find('input').at(0).simulate('change', {
//     target: { value }
//   });
//   expect(wrapper.props('values')).toBe({ name: value});
// });