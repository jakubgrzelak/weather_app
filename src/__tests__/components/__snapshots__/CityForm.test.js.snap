// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseForm correctly with expense data 1`] = `ReactWrapper {}`;

exports[`should render city form correctly 1`] = `"<Formik validate={[Function]} handleSubmit={[Function: handleSubmit]} displayName=\\"BasicForm\\" validationSchema={[undefined]} initialValues={{...}} initialStatus={[undefined]} onSubmit={[Function]} render={[Function]} validateOnChange={true} validateOnBlur={true} isInitialValid={false} enableReinitialize={false} />"`;
